### ISSUE TITLE
More dialog - phone number subscription status

### ISSUE THREAD

**Issue by whoabuddy** [COVID-19-electronic-health-system/Corona-tracker#537]
### Prerequisites 

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in QA
    * Followed all applicable steps in the TROUBLESHOOTING guide: TODO
    * Checked for relevant discussions on Discord: <https://discord.gg/pPERUuv>
    * Checked that your issue isn't already filed: <https://github.com/COVID-19-electronic-health-system/Corona-tracker/issues>

### Description

After entering a phone number into the more screen and hitting subscribe, there is no confimration, and accessing the screen again provides the same prompt.

### Steps to Reproduce

1. Log into the app
2. Click the more button
3. Enter the phone number
4. Click subscribe
5. Click the more button again

![20200416_115229_Phone_Number_Setting_Bug](https://user-images.githubusercontent.com/9038904/79507980-8f842400-7fed-11ea-84be-325ba4595daf.gif)

**Expected behavior:**

There should be a prompt confirming subscription (or unsubscription) of a phone number was successful.

There should be an indicator that a phone number is subscribed when opening the More screen.

**Actual behavior:**

There is no confirmation and now way to see that a phone number is subscribed.

**Reproduces how often:**

100%

### Additional Information

I wonder if it is easier to include the unsubscribe action as something you can click on with the phone number, such that if a phone number is subscribed, it shows on the list, and clicking the X on that phone number will unsubscribe it.

**Comment by fallon7284** [comment]
I'd work on this but I'm not clear on what goes on with the subscription API calls... currently I get CORS errors whether I'm trying it running the app locally, the AWS site, the coronatrackerbeta site, (I think those two are the same, but that's not the point) so I can't really do anything.  What kind of response do we get from a successful subsription?  

What I would think you'd want to do is that you'd have a subscribedPhoneNumber key on state which you'd map to More and if you would only render 
![Screen Shot 2020-04-17 at 1 40 55 PM](https://user-images.githubusercontent.com/15807680/79598040-48f7fd80-80b1-11ea-87b1-3676c8a26660.png)
if you did not have a number subscribed.  If you did, you'd instead render something like `Phone number ${subscribedPhoneNumber} subscribed to text message alerts` along with the unsubscribe button.

**Comment by SomeMoosery** [comment]
I like the second suggestion of adding phone number to Gaia, and programmatically rendering text in the more component based on if the user has a subscribed phone number. Would you want to take a crack at this one?

We are currently calling a lambda function behind an API gateway on the backend.

**Comment by fallon7284** [comment]
@SomeMoosery I'll take a crack at this but I'd like to know if what I'm seeing with regard to CORS errors from that subscribe function are also what you see... It's strange, I saw CORS errors every time I used it, running local and also deployed, but I'm getting messages so it did subscribe me.

Do I expect to get a response back from that API call indicating success or failure?

**Comment by SomeMoosery** [comment]
![Screen Shot 2020-04-19 at 7 12 44 PM](https://user-images.githubusercontent.com/15767602/79702287-c79b9900-8271-11ea-9b83-dfcaf37ae31f.png)

So I'm not sure why those CORS issues are coming up, but I'd likely attribute it to the `subscribe` lambda in `Coronalert` not having the requisite allowances on the server side - it may be worth referencing this bug there as well.

I'm also seeing a failed POST, which should be expected here as I tried entering in an already-subscribed phone number, so perhaps when we get something like this we update the text in the `More` dialog.

**Comment by fallon7284** [comment]
Hey @SomeMoosery I've been away from this project for a few days here but I'm back again and I'm unable to make any progress on this issue until this CORS issue is worked out.  I never get a response from the server side, and without a response I can't know whether the subscribe/unsubscribe lamba has succeeded or failed.  I don't believe that the POST failure you saw is because it was an already-subscribed number, I see it every time.  Interestingly, it does still work, because I've gotten the texts.  But if I never get the response object back, there isn't anything I can do.

Is there any way you can try to fix the CORS issue?  I've looked at it but go is lost on me.

**Comment by SomeMoosery** [comment]
Apologies - if you're running it locally I've actually installed and have been using [this](https://chrome.google.com/webstore/detail/allow-cors-access-control/lhobafahddgcelffkeicbaginigeejlf?hl=en) Chrome extension (if you're on Chrome) which circumvents CORS issues and allows things to work properly

**Comment by fallon7284** [comment]
@SomeMoosery so just to check in with you on this, I've got this basically finished (sorry for the delay, been a bit busy) and it's working great.  I used the chrome extension you pointed me toward and it allowed me to get past those CORS errors and actually use the response from the lambda.  

BUT

we still have the same issue going forward.  None of this functionality will work when we deploy it unless the users are also using that chrome extension, or the CORS issue gets fixed on the server side.  I've got just a bit left to clean up here, but it should be ready to PR soon, but presumably we don't want to actually have this code go live until it can work properly with the backend.

Just wanted to get your thoughts on how to proceed.
